cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/ExtractableExpression.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.relational;
+
+/**
+ * Expression that can be used to derive a scalar value. This value can then be used to build a new valid expression
+ * using {@link ExpressionFactory#compile}
+ * Note that relations, engine and expression factories must come from the same {@link RelationalTranformContext}
+ * and are valid only for the time of the {@link RelationalTransform#transform} call.
+ */
+public interface ExtractableExpression<T> extends Expression {
+  /**
+   * Get the value contained in this expression. This value is safe to use when building expressions using
+   * {@link ExpressionFactory#compile}
+   * @return scalar value contained in this expression
```

Comment:
Please either return optional or declare which exception will be thrown if expression is invalid